[docs] Update Functions package content (#8005)
apache/pulsar: site2/docs/functions-package.md

sidebar_label: "How-to: Package"
---

> You can package Pulsar functions in Java, Python, and Go. Packaging a window function in Java is the same as [packaging a function in Java](#java).

Reviewer: a window function or the window function?

> **Note**    

> > Currently, window function is not available in Python and Go.

Reviewer: ```suggestion
> Currently, the window function is unavailable in Python and Go.
```

## Prerequisite

> Before running a Pulsar function, you need to start Pulsar. You can [run a standalone Pulsar in Docker](getting-started-docker.md), or [run Pulsar cluster in Kubernetes](getting-started-helm.md).

Reviewer: run pulsar cluster or run pulsar?

To check whether the Docker image starts, you can use the `docker ps` command.

## Java 

To package a function in Java, complete the following steps.

> 1. Create a new maven project with a pom file. In the following code sample, `mainClass` is your package name.

Reviewer: it should be Maven

Author: @wolfstudy please confirm, thank you.

        }
    }
    ```
 

>      For the package imported, you can use one of the following interfaces:

Reviewer: ```suggestion
     For the imported package, you can use one of the following interfaces:
```

    The main difference between the two interfaces is that the `org.apache.pulsar.functions.api.Function` interface provides the context interface. When you write a function and want to interact with it, you can use context to obtain a wide variety of information and functionality for Pulsar Functions.

    The following example uses `org.apache.pulsar.functions.api.Function` interface with context.

    ```
    package org.example.functions;
    import org.apache.pulsar.functions.api.Context;
    import org.apache.pulsar.functions.api.Function;
    
    import java.util.Arrays;
    public class WordCountFunction implements Function<String, Void> {
       // This function is invoked every time a message is published to the input topic
        @Override
        public Void process(String input, Context context) throws Exception {
           Arrays.asList(input.split(" ")).forEach(word -> {
               String counterKey = word.toLowerCase();
               context.incrCounter(counterKey, 1);
            });
           return null;
       }
     }
     ```

3. Package the Java function.

    ```bash
    mvn package
    ```

>     After the Java function is packaged, a `target` directory is created automatically. Open the `target` directory to see if there is a jar package similar to `java-function-1.0-SNAPSHOT.jar`.

Reviewer: ```suggestion
    After the Java function is packaged, a `target` directory is created automatically. Open the `target` directory to check if there is a JAR package similar to `java-function-1.0-SNAPSHOT.jar`.
```

    ...
    ```

### ZIP file

> To package a function with **ZIP file** in Python, complete the following steps.

Reviewer: ```suggestion
To package a function with the **ZIP file** in Python, complete the following steps.
```

    }
    ```

    You can use context to connect to the Go function.
    
    ```
    if fc, ok := pf.FromContext(ctx); ok {
        fmt.Printf("function ID is:%s, ", fc.GetFuncID())
        fmt.Printf("function version is:%s\n", fc.GetFuncVersion())
     }
     ```

    When writing a Go function, remember that
    - In `main()`, you **only** need to register the function name to `Start()`. **Only** one function name is received in `Start()`. 

>     - Go function uses Go reflection based on the received function name to verify whether the parameter list and returned value list implemented are correct. The parameter list and returned value list specified **must be** one of the following sample functions:

Reviewer: I think this sentence needs to be refined:
whether the parameter list and returned value list implemented are correct.
The parameter list and returned value list specified **must be** one of the following sample functions:

Author: nice catch, I can simplify it, thanks for your reminder.
